### PR TITLE
Refactor the whole application kernel

### LINKED ISSUE #395: Issue when using `import` from a git repository
When utilizing the `import` statement to import helpers from a git repository, the `import` functionality may not behave as expected when updating the repository.

The `import` operation involves downloading the repository and storing it in the `vendor` directory of the project. However, when pushing fixes and updates the repository, the `import` statement not fetching the latest version.

This issue is not strictly related to `castor` but rather appears to be more of a `composer` problem.

#### Steps to Reproduce the Issue:

```php
import('package://theod02/castor-class-task', source: [
    'url' => 'https://github.com/TheoD02/castor-class-task.git',
    'type' => 'git',
    'reference' => 'main',
]);
```

```bash
# Will download remotes and all files
castor test
```

After making a mistake and pushing a fix to the repository `castor-class-task`:

```bash
# Should fetch the new version
castor --update-remotes
```

However, in this scenario, Composer have cached the previous version. To resolve this, you may need to remove the `vendor` directory and run `composer clear-cache`, then re-run `castor` to import the remotes again.

Is the expected behavior ?

### PR DESCRIPTION
**Pull request by lyrixx**
* Big refactoring, cleaner code
* Allow remote import in remote import (**BUT THIS IS NOT RECOMMENDED**)
* Fix issue with fingerprint when fetching import ; fix #395

### CODE REVIEW

**Review by pyrech — COMMENTED** [review]
Nice refacto :yellow_heart:

File: src/Import/Importer.php [hunk 1]
```diff
@@ -72,34 +77,30 @@ public function import(string $path, ?string $file = null, ?string $version = nu
...
 
             foreach ($files as $file) {
-                castor_require($file->getPathname());
+                $this->importFile($file->getPathname());
             }
         }
     }
 
-    public function require(string $path): void
+    /** @internal */
```
[the remark this answers is not in the document]
Comment (lyrixx) [reply src/Import/Importer.php:86]: Indeed, my mistake

**Review by lyrixx — COMMENTED** [review]
Okay, the PR is now ready for review

**Review by pyrech — APPROVED** [review]

**Comment by lyrixx** [comment]
> Is there any way to forbid it? I think it could lead to security issues (someone publish a castor random task that sounds useful, but this task import another one (repeat N times) and the last one contain malicious code.

Indeed, but it's already the case with a "simple" import. I mean, this PR does not allow more security issue than the one already present

**Pull request merged**